db/active_replicator_checkpointer.go

```diff
@@ -388,6 +388,51 @@ func (r *replicationCheckpoint) Copy() *replicationCheckpoint {
 	}
 }
 
+// fetchNamedCollectionCheckpoints will:
+// - fetch the remote and local checkpoints off the checkpointer object
+// - increments miss count if the remote and local checkpoints are not the same
+// - increment hit count if successfully parse sequence ID
+// - increment miss count if error during parse sequence ID or if checkpoint sequence miss
+func (c *Checkpointer) fetchNamedCollectionCheckpoints() error {
+	localCheckpoint, err := c.getLocalCheckpoint()
+	if err != nil {
+		return err
+	}
+
+	base.DebugfCtx(c.ctx, base.KeyReplicate, "got local checkpoint: %v", localCheckpoint)
+	c.lastLocalCheckpointRevID = localCheckpoint.Rev
+
+	remoteCheckpoint, err := c.getRemoteCheckpoint()
+	if err != nil {
+		return err
+	}
+	base.DebugfCtx(c.ctx, base.KeyReplicate, "got remote checkpoint: %v", remoteCheckpoint)
```

Comment:
This code is duplicated by the `_initCollections` code and I'm reluctant to call it twice. We have the checkpoint from `apr._initCollections`.

I'd either do this ticket two ways - 

1. in `_startPullWithCollections` and `_startPushWithCollections`, just put hit or miss stats right before we do `subChanges` based on whether `since` is empty string or not for collections code
2. Implement https://issues.couchbase.com/browse/CBG-2788 by separating how we call `getRemoteCheckpoint` as a string value that is passed into this function (it will either come from a direct call to `getRemoteCheckpoint` or from `replicationCheckpoint` from `_initCollections`, and call the same function for the default and non default.

I could see it is possible if we test for misses based on whether local checkpoint and remote checkpoint misses we won't hit this. In general, the miss is whether or not the checkpoint is present on the remote.

I'd be inclined to implement (1) as a stop gap simpler, and then address (2) as a separate ticket,  though some tests might have to _still_ be skipped.